l10n_ec_credit_note/models/account_move.py

```diff
@@ -0,0 +1,119 @@
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_ec_type_credit_note = fields.Selection(
+        [("discount", "Discount"), ("return", "Return")],
+        string="Credit note type",
+        default="discount",
+    )
+
+    def action_reverse(self):
+        action = super().action_reverse()
+        if (
+            self.company_id.account_fiscal_country_id.code == "EC"
+            and self.move_type == "out_invoice"
+        ):
+            action["views"] = [
+                (
+                    self.env.ref(
+                        "l10n_ec_credit_note.view_account_invoice_refund_sale"
+                    ).id,
+                    "form",
+                )
+            ]
+        return action
+
+    def _reverse_moves(self, default_values_list=None, cancel=False):
+        reverse_moves = super()._reverse_moves(
+            default_values_list=default_values_list, cancel=cancel
+        )
+        if (
+            self.company_id.account_fiscal_country_id.code == "EC"
+            and reverse_moves.filtered(
+                lambda x: x.l10n_ec_type_credit_note and x.move_type in ["out_refund"]
+            )
+        ):
+            rev_mov_fil = reverse_moves.filtered(
+                lambda x: x.l10n_ec_type_credit_note and x.move_type in ["out_refund"]
+            )
+            rev_mov_lines = rev_mov_fil.with_context(skip_invoice_sync=cancel).mapped(
+                "line_ids"
+            )
+            for reverse_move_line in rev_mov_lines.filtered(
+                lambda x: x.display_type == "product"
+            ):
+                reverse_move_line.account_id = self._get_account_product_line(
+                    reverse_move_line
+                )
+        return reverse_moves
```

Comment:
Can this code be handled at a different time or in another function (e.g., on functions `copy` or `copy_data`)? I see it as risky because when the `super call` occurs, the record might already be `posted`, and it could be problematic to try changing the `account` if the document is already posted.
https://github.com/odoo/odoo/blob/96b3ffc2394be46a4f7a44b7a618a7ee8f2c594f/addons/account/models/account_move.py#L3768

Reply:
Hi 

I tried to override the method that is triggered when using the "Reverse and Create Invoice" button in the credit note wizard, as at that moment, the invoice is only duplicated, and the data is changed.

https://github.com/odoo/odoo/blob/96b3ffc2394be46a4f7a44b7a618a7ee8f2c594f/addons/account/models/account_move.py#L3756

However, when the company uses Anglo-Saxon accounting, the same method I'm using is overridden to handle the case of adding the required logic.

https://github.com/odoo/odoo/blob/96b3ffc2394be46a4f7a44b7a618a7ee8f2c594f/addons/stock_account/models/account_move.py#L27

In which other method could I override it so that when it's a credit note for a discount, the Anglo-Saxon accounts are not applied?

Reply:
Hi @celm1990 

You were right; I was confused in my initial suggestion. As you pointed out, trying to change the account after the document is already posted could be risky.

I’ve updated the code to handle this logic in the copy_data method, ensuring that the necessary modifications are made at the correct time, before the record is duplicated and the move is posted.

The changes are included in the latest commit. Could you please review it to confirm it now meets the requirements?

Thanks for the feedback.